### DISCUSSION TITLE
posts/multi-arch-k3s-rpi/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#239] [Comments]
# posts/multi-arch-k3s-rpi/

Building a Multi-architecture CPU Kubernetes cluster is easier than you think with k3s. In this video we’ll build a Raspberry Pi 4 with an ARM CPU and add it to our existing x86 x64 amd64 CPU Kubernetes cluster. Our foundation will be Ubuntu for ARM, then we’ll add k3s, and then join it to our cluster. We’ll also discuss how this works with Docker images built for specific CPU types. We’ll also talk about some build configurations and requirements for your Pi.

https://technotim.live/posts/multi-arch-k3s-rpi/

**Comment by elronzo** [comment]
There is no need for creating an architecture-specific label.  You could use:

```
  nodeSelector:
    kubernetes.io/arch: arm64
```
